packages/xo-server/src/index.js

```diff
@@ -283,12 +284,17 @@ async function registerPlugin(pluginPath, pluginName) {
   } = plugin
   let instance
 
-  const handleFactory = factory =>
+  const config = this._config
+  const handleFactory = async factory =>
     typeof factory === 'function'
       ? factory({
+          config: merge(
+            await readFile(`${pluginPath}/config.toml`).then(TOML.parse, noop),
+            config?.plugins[pluginName]
```

Comment:
Is `config.plugins` which can be `undefined` because it's a new entry and it isn't defined in the default configuration :+1: 
So it'll be `config.plugins?.[pluginName]`